src/App.js

```diff
@@ -14,11 +16,87 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+export class App extends Component {
+  state = {
+    goods: [...goodsFromServer],
+    visible: false,
+  };
+
+  startApp = () => {
+    this.setState({ visible: true });
+  };
+
+  reverseList = () => {
+    this.setState(state => ({
+      goods: [...state.goods].reverse(),
+    }));
+  }
+
+  sortList = () => {
+    this.setState(state => ({
+      goods: [...state.goods].sort((a, b) => a.localeCompare(b)),
+    }));
+  }
+
+  resetList = () => {
+    this.setState({
+      goods: [...goodsFromServer],
+    });
+  }
+
+  sortLength = () => {
+    this.setState(state => ({
+      goods: [...state.goods].sort((a, b) => a.length - b.length),
+    }));
+  }
+
+  render() {
+    const { goods, visible } = this.state;
+
+    return (
+      <>
+        <button
```

Comment:
```suggestion
        {!visible && <button
```
Use conditional rendering instead of setting `display: none`